src/pocketmine/Player.php

```diff
@@ -2864,6 +2864,10 @@ public function handleDropItem(DropItemPacket $packet) : bool{
 			// Windows 10 Edition drops the contents of the crafting grid on container close - including air.
 			return true;
 		}
+		
+		if($this->gamemode === self::SPECTATOR) {
```

Comment:
Code readability and ease of development takes priority over saving a couple of microseconds here and there. If you think quibbling over things like this will improve performance in any noticeable manner, you're wasting your time.